### DISCUSSION TITLE
handling errors from mutate()

### DISCUSSION THREAD

**Discussion by jonathanstanley** [vercel/swr#810] [Q&A] [answered]
I appreciate the library!  From the docs, it isn't clear if there is a proper way to handle exceptions thrown during `mutate()`

Suppose I want a user's action to trigger a mutation for a specific key (keeping a static key enables sharing between different components).  I may want the resulting data or error kept within the swr.  For example:
```javascript
let mySWR = useSWR("myKey");
mySWR.mutate(fetchFromAPI).catch((e) => {});
console.log(mySWR.data, mySWR.error)
//if `fetchFromApi` returns a result, `mySWR.data` should have the result and error is undefined
//if `fetchFromApi` throws an error, `mySWR.error` should have the error text and data is undefined
```
**But that doesn't quite work.**  Here is my example: https://codesandbox.io/s/swr-mutate-error-user-example-mxp14

Some problems:

1. Depending on where`console.log(mySWR)` runs, it may show the correct error, or it may hold a random `SyntaxError: Unexpected token < in JSON at position 0`
2. If I try to conditionally show the error or data, SWR starts running excessive re-validations

Thoughts ?

**Comment by shuding** [comment, marked as answer]
Thank you for your codesandbox link! The cause of `SyntaxError: Unexpected token < in JSON at position 0` is that your `useSWR` doesn't have `fetcher` passed in. So it uses the default `fetch` function, which can't fetch the key since it's not a URL.

You can do this:

```jsx
useSWR('key', null)
```

to avoid that behavior.

**Reply by jonathanstanley** [reply]
thank you
